### PR TITLE
Created calendar

### PR DESCRIPTION
**Pull request by natalia-matash**
- [DEMO LINK](https://natalia-matash.github.io/layout_calendar/)
- [TEST REPORT LINK](https://natalia-matash.github.io/layout_calendar/report/html_report/)

### CODE REVIEW

**Comment by natalia-matash** [comment]
When I did loops @each and @for, when I put it into .calendar {} or in &__item {} they didn't work.

File: src/index.html [hunk 1]
```diff
@@ -1,12 +1,44 @@
...
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>Calendar</title>
     <link rel="stylesheet" href="./styles/main.scss">
   </head>
-  <body>
-    <h1>Calendar</h1>
+  <body class="page__contant">
+    <div class="calendar calendar--start-day--sun calendar--month-length--31">
+      <div class="calendar__item calendar__item--days--1"></div>
```
Comment (loralevitska) [inline src/index.html]: Don't add modifier for each day

File: src/styles/blocks/calendar.scss [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      }
+
+      100% {
+        transform: translateY(-20px);
+      }
+    }
+  }
+}
+
+@for $i from 1 to 4 {
```
Comment (loralevitska) [inline src/styles/blocks/calendar.scss]: from 28 to 31

**Review by loralevitska — CHANGES_REQUESTED** [review]

**Comment by loralevitska** [comment]
> When I did loops @each and @for, when I put it into .calendar {} or in &__item {} they didn't work.

This task is designed for maximum practice cycles. So write down your solution and ask for advice in fe_chat and tag mentors so we can help

File: src/styles/utils/_variables.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+$days-amount: 31;
+
+$week-days:
+  "mon" 0,
+  "tue" 101px,
+  "wed" 202px,
+  "thu" 303px,
+  "fri" 404px,
+  "sat" 505px,
+  "sun" 606px;
```
Comment (agniya-i) [inline src/styles/utils/_variables.scss]: try to find a way to calculate this value depends on day `width`

File: src/styles/blocks/calendar.scss [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  display: flex;
+  gap: 1px;
+  flex-wrap: wrap;
+  max-width: 706px;
+
+  &__item {
+    display: flex;
+    position: relative;
+    width: 100px;
+    height: 100px;
```
Comment (agniya-i) [inline src/styles/blocks/calendar.scss]: add variables for those values

File: src/styles/blocks/calendar.scss [hunk 5]
```diff
@@ -0,0 +1,75 @@
+@import "../utils/variables";
+
+.calendar {
+  padding: 20px 10px;
+  display: flex;
+  gap: 1px;
```
Comment (agniya-i) [inline src/styles/blocks/calendar.scss]: add variable

**Review by agniya-i — CHANGES_REQUESTED** [review]

File: src/styles/blocks/calendar.scss [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    @keyframes translate {
+      0% {
+        transform: translateY(0);
+      }
+
+      100% {
+        transform: translateY(-20px);
+      }
+    }
+  }
```
Comment (SanyaBratashchuk) [inline src/styles/blocks/calendar.scss]: use transition instead animation

File: src/styles/utils/_variables.scss [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+$item-length: 100px;
+
+$week-days:
+  "mon" 0,
+  "tue" #{$item-length + $gap},
+  "wed" #{$item-length * 2 + $gap * 2},
+  "thu" #{$item-length * 3 + $gap * 3},
+  "fri" #{$item-length * 4 + $gap * 4},
+  "sat" #{$item-length * 5 + $gap * 5},
+  "sun" #{$item-length * 6 + $gap * 6};
```
Comment (SanyaBratashchuk) [inline src/styles/utils/_variables.scss]:
```suggestion
  "tue" #{$item-length + $gap},
  "wed" #{$item-length * 2 + $gap * 2},
  "thu" #{$item-length * 3 + $gap * 3},
  "fri" #{$item-length * 4 + $gap * 4},
  "sat" #{$item-length * 5 + $gap * 5},
  "sun" #{$item-length * 6 + $gap * 6};
```
move calculations inside loop

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

**Review by marina-tilniak — APPROVED** [review]

**Pull request closed without merging**